plugins/font/plugin.js

```diff
@@ -1,45 +1,75 @@
-﻿/**
+/**
  * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
  * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
  */
 
 ( function() {
-	function addCombo( editor, comboName, styleType, lang, entries, defaultLabel, styleDefinition, order ) {
+	function addCombo( editor, comboName, styleType, lang, entries, defaultLabel, configStyleDefinition, order ) {
 		var config = editor.config,
-			style = new CKEDITOR.style( styleDefinition );
+			allowedAndRequiredContent = new CKEDITOR.style( configStyleDefinition ),
+			commandName = comboName.slice( 0, 1 ).toLowerCase() + comboName.slice( 1 ),
+			preparedStylesAndNames = _prepareStylesAndNames( {
+				entries: entries,
+				styleType: styleType,
+				configStyleDefinition: configStyleDefinition
+			} ),
+			styles = preparedStylesAndNames.styles,
+			names = preparedStylesAndNames.names;
+
+		editor.addCommand( commandName , {
+			contextSensitive: true,
+			exec: function( editor, data ) {
+				if ( editor.readOnly ) {
+					return;
+				}
 
-		// Gets the list of fonts from the settings.
-		var names = entries.split( ';' ),
-			values = [];
+				var newStyle = data.newStyle,
+					oldStyle = data.oldStyle,
+					range = editor.getSelection().getRanges()[ 0 ],
+					isRemove = newStyle === undefined;
 
-		// Create style objects for all fonts.
-		var styles = {};
-		for ( var i = 0; i < names.length; i++ ) {
-			var parts = names[ i ];
+				// Prevent of throw error when situation from this bug appear (#1116).
+				if ( !oldStyle && !newStyle ) {
+					return;
+				}
 
-			if ( parts ) {
-				parts = parts.split( '/' );
+				// If the range is collapsed we can't simply use the editor.removeStyle method
+				// because it will remove the entire element and we want to split it instead.
+				if ( oldStyle && range.collapsed ) {
+					_splitElementOnCollapsedRange( {
+						editor: editor,
+						range: range,
+						style: oldStyle
+					} );
+				}
 
-				var vars = {},
-					name = names[ i ] = parts[ 0 ];
+				if ( isRemove ) {
+					editor.removeStyle( oldStyle );
+				} else {
+					if ( oldStyle ) {
+						editor.removeStyle( oldStyle );
+					}
 
-				vars[ styleType ] = values[ i ] = parts[ 1 ] || name;
+					editor.applyStyle( newStyle );
+				}
+			},
 
-				styles[ name ] = new CKEDITOR.style( styleDefinition, vars );
-				styles[ name ]._.definition.name = name;
-			} else {
-				names.splice( i--, 1 );
+			refresh: function( editor, path ) {
+				if ( !allowedAndRequiredContent.checkApplicable( path, editor, editor.activeFilter ) ) {
+					this.setState( CKEDITOR.TRISTATE_DISABLED );
+				} else {
+					this.setState( CKEDITOR.TRISTATE_OFF );
+				}
 			}
-		}
+		} );
 
 		editor.ui.addRichCombo( comboName, {
```

Comment:
Interestingly, you can still provide `command` for richcombo and it will be correctly added as a command UI element:
https://github.com/ckeditor/ckeditor4/blob/066ab2b99e4c634a61bb1d6f292ab297cfe464f0/core/ui.js#L92
But not sure if it does anything special as rich combo doesn't use it. Still, nice to keep it unified with the rest of the UI items, at least partially.